Document dependencies better (#1249)
trufflesuite/trufflesuite.com: src/docs/truffle/getting-started/installation.md

---
# Installation

## Requirements

> * [NodeJS v12, or later](#install-nodejs)

Reviewer: Fixed.

* Windows, Linux, or macOS

## Install NodeJS

### Linux and macOS

NPM recommends [installing Node.js with a version manager (NVM)](https://npm.github.io/installation-setup-docs/installing/using-a-node-version-manager.html)

> to avoid permission errors when installing globally.

Reviewer 2: Also should probably call out somewhere you might run into issues down the line if you install truffle with `sudo` which some users do just cause they think it's easier

> to avoid permission errors when installing globally.

Reviewer: @emilyJLin95, what sort of issues are users running into by installing using `sudo`? Permissioning errors?

> to avoid permission errors when installing globally.

Reviewer 2: > Permissioning errors
Yep! In general, it's highly likely users will run into issues down the line

**Important**: Truffle requires `node-gyp` for compiling native addon modules for Node.js.  Node-gyp in
turn has [dependencies](https://github.com/nodejs/node-gyp) which may be missing, causing the Truffle install to fail.
Truffle recommends installing the following to avoid getting dependency errors:

* macOS only: `xcode-select --install`
* Linux (Ubuntu) only: `sudo apt install make g++`

### Windows

Truffle recommends using the installer available from the [nodejs site](https://nodejs.org/en/download/).

Ensure you select **Automatically install the necessary tools...** during the install to install the
required Visual Studio build tools, Python, and Chocolately package manager.

![Architecture](/img/docs/truffle/installation/windows-nodejs.png)

> ## Install Truffle

Reviewer 2: Not sure what best practices for docs are, but would it be worthwhile to include a test see that truffle is installed by using truffle version?

Unfortunately, truffle spits out a long laundry list installation warnings, which a lot of users get confused by and think installation was unsuccessful because of it

Reviewer: Fixed

---
# Installation

## Requirements

* [Node.js v12 - v16](#install-nodejs)
* Windows, Linux, or macOS

## Install Node.js

### Linux and macOS

Node Package Manager (NPM) recommends [installing Node.js with Node Version Manager (`nvm`)](https://node.dev/post/installing-node-js-tutorial-using-nvm-on-mac-os-x-and-ubuntu)
to avoid permission errors when installing globally.

1. Use `curl` or `wget` to install `nvm`:

    === "curl"

        ```bash
        curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
        ```

    === "wget"

        ```bash
        wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
        ```

1. Confirm that `nvm` installed correctly. You may need to reload the terminal for
    the command to work:

    ```bash
    nvm --version
    ```
 

> 1. Truffle requires `node-gyp` for compiling native addon modules for Node.js.  Node-gyp in
>     turn has [dependencies](https://github.com/nodejs/node-gyp) which may be missing, causing the Truffle install to fail.
>     Truffle recommends installing the following to avoid getting dependency errors:

Reviewer 3: ```suggestion
1. Truffle requires `node-gyp` for compiling native add-on modules for Node.js. 
    Truffle recommends installing the following `node-gyp` [dependencies](https://github.com/nodejs/node-gyp) to avoid errors when installing Truffle:
```

    === "macOS"

        ```bash
        xcode-select --install
        ```

    === "Linux (Ubuntu)"

        ```bash
        sudo apt install make g++
        ```

> 1. Use `nvm` to install a compatible version of NodeJS. For example, to install NodeJS v15, run:

Reviewer 3: ```suggestion
1. Use `nvm` to install a compatible version of Node.js. For example, to install Node.js v15, run:
```

    ```bash
    nvm install 15
    ```

1. Confirm that Node.js has been installed correctly by running `node --version`.

### Windows

> Truffle recommends using the installer available from the [nodejs site](https://nodejs.org/en/download/).

Reviewer 3: ```suggestion
Truffle recommends using the installer available from the [Node.js site](https://nodejs.org/en/download/).
```

Ensure you select **Automatically install the necessary tools...** during the install to install the
required Visual Studio build tools, Python, and Chocolately package manager.

![Architecture](/img/docs/truffle/installation/windows-nodejs.png)

## Install Truffle

<p class="alert alert-warning">
<i class="far fa-exclamation-triangle"></i> <strong>Warning</strong>: Avoid using the <code>sudo</code> command when installing Truffle, this can cause permission errors.
</p>

In a terminal, use NPM to install Truffle:

```bash
npm install -g truffle
```

> You may receive a list of warnings during installation. To confirm that truffle was installed correctly,

Reviewer 3: ```suggestion
You may receive a list of warnings during installation. To confirm that Truffle was installed correctly,
```
